doofinder-for-woocommerce/includes/class-update-manager.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+namespace Doofinder\WC;
+
+use Doofinder\WC\Helpers\Helpers;
+use Doofinder\WC\Settings\Settings;
+
+defined('ABSPATH') or die;
+
+class Update_Manager
+{
+	private static $logger = NULL;
+
+	/**
+	 * Checks and performs any pending DB updates.
+	 * 
+	 * @since 1.5.30
+	 */
+	public static function check_updates($plugin_version)
+	{
+		$db_version = Settings::get_plugin_version();
+		self::log("Check updates from $db_version to $plugin_version");
+
+		if (empty($plugin_version)) {
+
+			self::log("invalid plugin version: $plugin_version");
+
+			return false;
+		}
+		$current_normalized_version = self::normalize_plugin_version(Settings::get_plugin_version());
+		$plugin_normalized_version = self::normalize_plugin_version($plugin_version);
+
+		for ($version = $current_normalized_version + 1; $version <= $plugin_normalized_version; $version++) {
+			$version_number = str_pad($version, 6, '0', STR_PAD_LEFT);
+			$update_function = 'update_' . $version_number;
+
+			self::log("check if the update  $update_function exists.");
+			if (method_exists(Update_Manager::class, $update_function)) {
+				self::log("Executing $update_function update...");
+				$result = NULL;
+				try {
+					$result = call_user_func(array(Update_Manager::class, $update_function));
+					if (!$result) {
+						//Throw an empty error to add a default notice
+						throw new \Exception("");
```

Comment:
Does it make sense this exception that we are going to capture in the general catch?

Reply:
I've made some changes and removed this exception